### PR TITLE
Collect DB latency metrics

### PR DESCRIPTION
**Pull request by omerfirmak**
Should help us compare storages of different machines.

### CODE REVIEW

File: node/metrics.go [hunk 1]
```diff
@@ -140,7 +160,10 @@ func makeSyncMetrics(syncReader sync.Reader, bcReader blockchain.Reader) sync.Ev
 		OnSyncStepDoneCb: func(op string, blockNum uint64, took time.Duration) {
 			opTimerHistogram.WithLabelValues(op).Observe(took.Seconds())
 			if op == sync.OpStore {
+				fmt.Println("store", blockNum, took)
 				blockCount.Inc()
+			} else if op == sync.OpVerify {
+				fmt.Println("verify", blockNum, took)
```
[the remark this answers is not in the document]
Comment (omerfirmak) [reply node/metrics.go:166]: nope, my bad.

File: db/event_listener.go [hunk 2]
```diff
@@ -1,15 +1,17 @@
 package db
 
+import "time"
+
 type EventListener interface {
-	OnIO(write bool)
+	OnIO(write bool, took time.Duration)
```
Comment (IronGauntlets) [inline db/event_listener.go:6]: `duration` would be better name instead of `took`

Comment (omerfirmak) [reply db/event_listener.go]: done

File: node/metrics.go [hunk 3]
```diff
@@ -11,22 +12,40 @@ import (
...
+		500,
+		1000, // 1ms
+		2000,
+		3000,
+		4000,
+		5000,
+		10000,
+		50000,
+		math.Inf(0),
+	}
```
Comment (IronGauntlets) [inline node/metrics.go:30]: Shouldn't buckets be either linear or exponential? this is neither, also before going to infinity, a 1-second bucket would be nice, right?

**Review by IronGauntlets — APPROVED** [review]

File: node/metrics.go (hunk 3 above)
Comment (omerfirmak) [reply node/metrics.go:30]: > Shouldn't buckets be either linear or exponential?

I don't think that is a requirement. But I don't really know if there is a correct set of values. I set these with by trial and error on my local SSD mainly. We can always update them if it is not making sense for the storage options cloud providers are offering. 

> a 1-second bucket would be nice, right?

adding that as well.

**Pull request merged**